kv/kvserver: TestWriteLoadStatsAccounting failed

kv/kvserver.TestWriteLoadStatsAccounting [failed](https://mesolite.cluster.engflow.com/invocations/default/0e0afdb7-2d01-4233-a344-b62c5a50b4c3?testReportRun=1&testReportShard=48&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.4 @ [dba21b616ff89f4718ce04d4b47a9baa07127a03](https://github.com/cockroachdb/cockroach/commits/dba21b616ff89f4718ce04d4b47a9baa07127a03):


```
=== RUN   TestWriteLoadStatsAccounting
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestWriteLoadStatsAccounting1292535136
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:561: DRPC is enabled (override by TestingGlobalDRPCOption)
    replica_rankings_test.go:324: 
        	Error Trace:	pkg/kv/kvserver/replica_rankings_test.go:324
        	Error:      	"0" is not greater than or equal to "9"
        	Test:       	TestWriteLoadStatsAccounting
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestWriteLoadStatsAccounting1292535136
--- FAIL: TestWriteLoadStatsAccounting (1.99s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=48</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWriteLoadStatsAccounting.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58982

--- a/pkg/kv/kvserver/replica_rankings_test.go
+++ b/pkg/kv/kvserver/replica_rankings_test.go
@@ -304,6 +304,13 @@ func TestWriteLoadStatsAccounting(t *testing.T) {
 			lhRepl.Desc().Replicas().Descriptors(),
 			lhRepl.GetLeaseAppliedIndex(),
 		))
+		// Wait for raftMu on the followers to ensure that handleRaftReady has
+		// finished, including the load stats recording that happens after the
+		// applied index bump. See #161600.
+		followerRepl1.raftMu.Lock()
+		followerRepl1.raftMu.Unlock() //lint:ignore SA2001 empty critical section
+		followerRepl2.raftMu.Lock()
+		followerRepl2.raftMu.Unlock() //lint:ignore SA2001 empty critical section
 
 		requestsAfter := lhRepl.loadStats.TestingGetSum(load.Requests)
 		lhWritesAfter := lhRepl.loadStats.TestingGetSum(load.WriteKeys)

--- a/pkg/kv/kvserver/replica_rankings_test.go
+++ b/pkg/kv/kvserver/replica_rankings_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/storageutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -214,6 +215,11 @@ func assertGreaterThanInDelta(t *testing.T, expected float64, actual float64, de
 func TestWriteLoadStatsAccounting(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	// This test is known to flake. E.g. in #160265, a request raced in after
+	// clearing the load stats and before asserting they are 0.
+	skip.UnderDuress(t, "timing sensitive")
+
 	ctx := context.Background()
 
 	args := base.TestClusterArgs{

